go/obscuronode/host/host.go

```diff
@@ -18,13 +16,26 @@ import (
 
 const ClientRPCTimeoutSecs = 5
 
+// todo - this has to be replaced with a proper cfg framework
 type AggregatorCfg struct {
 	// duration of the gossip round
 	GossipRoundDuration uint64
 	// timeout duration in seconds for RPC requests to the enclave service
 	ClientRPCTimeoutSecs uint64
 }
 
+type P2PCallback interface {
+	P2PGossipRollup(r obscurocommon.EncodedRollup)
+	P2PReceiveTx(tx nodecommon.EncryptedTx)
+}
+
+type ObscuroNetwork interface {
```

Comment:
Why not just call this P2P? There's a mash up atm - the file and impl are called P2P, and main refers to it as P2P, but then we call it ObscuroNetwork in a few places too. Confusing I think.